src/settings/SettingsStore.ts

```diff
@@ -257,6 +257,11 @@ export default class SettingsStore {
         return SETTINGS[settingName].isFeature;
     }
 
+    public static isEnabled(settingName: string) {
+        if (!SETTINGS[settingName]) return false;
+        return SETTINGS[settingName].controller ? !SETTINGS[settingName].controller.settingDisabled : true;
+    }
+
```

Comment:
This isn't how we should introduce `isEnabled` imo - it's not entirely clear why this exists.

Instead, let's pass some more information through the constructor of the UIFeature controller, like what the "disabled" value should be and (if needed) information about what the inverted setting is.

Reply:
> it's not entirely clear why this exists.

It exists as a homogeneous wrapper around any settingName which checks whether it is disabled for any reason, such as if a UI Feature flag says so. Its a more data-driven approach then the code which predates it:

https://github.com/matrix-org/matrix-react-sdk/pull/5216/files#diff-c3b4838ca1cecfdde01b1a1376530b8fL141-L143

> Instead, let's pass some more information through the constructor of the UIFeature controller, like what the "disabled" value should be and (if needed) information about what the inverted setting is.

Sure but then accessing that field on that type of controller will mean a lot of duplicated code all around the place as not all settings have that controller and no other controllers expose that field.

Reply:
the store shouldn't need to reach into the controller?

Reply:
I don't think I'm seeing the solution you're proposing as an alternative

Reply:
Essentially this:
```
diff --git a/src/settings/controllers/UIFeatureController.ts b/src/settings/controllers/UIFeatureController.ts
index ed6598a6e8..e82f55d5e8 100644
--- a/src/settings/controllers/UIFeatureController.ts
+++ b/src/settings/controllers/UIFeatureController.ts
@@ -23,10 +23,12 @@ import SettingsStore from "../SettingsStore";
  * UI feature is disabled. If the UI feature is enabled, the setting value
  * is unchanged.
  *
- * Settings using this controller are assumed to return `false` when disabled.
+ * Settings using this controller are assumed to return `false` by default
+ * when disabled. This can be changed with the disabledValue constructor
+ * argument.
  */
 export default class UIFeatureController extends SettingController {
-    public constructor(private uiFeatureName: string) {
+    public constructor(private uiFeatureName: string, private disabledValue: any = false) {
         super();
     }

@@ -38,7 +40,7 @@ export default class UIFeatureController extends SettingController {
     ): any {
         if (!SettingsStore.getValue(this.uiFeatureName)) {
             // per the docs: we force a disabled state when the feature isn't active
-            return false;
+            return this.disabledValue;
         }
         return null; // no override
     }
```

Because the SettingsStore is already aware of inverted settings, the controller just needs to return a value as though it wasn't inverted - the settings store will take care of the rest. Inverted settings must also be booleans, so this is considered safe.

Reply:
Er but I already have that below.

`isEnabled` is to allow filtering out the Settings which are disabled out of the Settings UI,

To avoid code like https://github.com/matrix-org/matrix-react-sdk/pull/5216/files#diff-c3b4838ca1cecfdde01b1a1376530b8fL141-L143

Reply:
can we put a bunch of documentation on `isEnabled` then?

Reply:
including the settings.md developer guide to explain what an enabled setting is